### PR TITLE
Make ResizeObserver global as it's a browser API

### PR DESCRIPTION
**Pull request by wffurr**
Following the guidance in the TypeScript handbook for declaration files for global APIs.  Instead of pretending that ResizeObserver is a module, declare it as a global and also export it on the window.  Ideally this would be in the DOM typings, but it's still in draft and only supported by Chrome.

I haven't tested because "npm run test" appears to be broken at head.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/global-d-ts.html
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by wffurr** [comment]
Not sure where else to post this, but I made a SO question about the out of date types-publisher module that I'm getting: https://stackoverflow.com/questions/57209407/npm-install-is-downloading-old-version-of-types-publisher-from-github

File: types/resize-observer-browser/index.d.ts [hunk 1]
```diff
@@ -1,25 +1,23 @@
 // Type definitions for non-npm package resize-observer-browser 0.1
 // Project: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver, https://developers.google.com/web/updates/2016/10/resizeobserver, https://wicg.github.io/ResizeObserver/
-// Definitions by: Chives <https://github.com/chivesrs>
+// Definitions by: Chives <https://github.com/chivesrs>, William Furr <https://github.com/wffurr>
```
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Could you use the multline syntax here please?
https://github.com/DefinitelyTyped/DefinitelyTyped#edit-an-existing-package

File: types/resize-observer-browser/test/resize-observer-browser-module.tests.ts [hunk 2]
```diff
@@ -1,5 +1,3 @@
-import { ResizeObserver } from "resize-observer-browser";
```
Comment (chivesrs) [inline types/resize-observer-browser/test/resize-observer-browser-module.tests.ts]: It looks like we don't need a module test at all, given there is no import here.
Let's rename this file to `resize-observer-browser-tests.ts` (as per https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package), and remove the global tests file.

Then we should merge in the `window.ResizeObserver` test into here.

File: types/resize-observer-browser/test/resize-observer-browser-module.tests.ts [hunk 3]
```diff
@@ -1,5 +1,3 @@
...
-
 function resizeObserverCreates(): void {
     const resizeObserver: ResizeObserver = new ResizeObserver((entries) => {
```
Comment (chivesrs) [inline types/resize-observer-browser/test/resize-observer-browser-module.tests.ts]: We should add a test in the scenario where the callback is complex and so is its own function.

Something like

```
function callback(entries: TheType): void {
  // no-op
}

const resizeObserver = new ResizeObserver(callback);
```

This way we know `TheType` is usable without needing to be imported.

**Review by chivesrs — CHANGES_REQUESTED** [review]

**Comment by chivesrs** [comment]
The lint command seems to work for me, I'm using `npm run lint resize-observer-browser` from the root directory (the one which contains `README.md`).

`npm test` seems to not work for me, but I've found the lint command does a much better job anyway.

File: types/resize-observer-browser/test/resize-observer-browser-module.tests.ts (hunk 2 above)
Comment (chivesrs) [reply types/resize-observer-browser/test/resize-observer-browser-module.tests.ts]: Oh, I think the file should be moved out of the `tests/` subdirectory as well if we do this.

**Review by chivesrs — CHANGES_REQUESTED** [review]

File: types/resize-observer-browser/index.d.ts [hunk 4]
```diff
@@ -1,25 +1,23 @@
...
-declare global {
-    interface Window {
-        ResizeObserver: typeof ResizeObserver;
-    }
+interface Window {
+    ResizeObserver: typeof ResizeObserver;
 }
 
-export class ResizeObserver {
+declare class ResizeObserver {
```
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Oh, `.d.ts` files implicitly add declare before every single item. Do you need this here? Or does everything work without the declare keyword?

**Review by chivesrs — CHANGES_REQUESTED** [review]

File: types/resize-observer-browser/index.d.ts (hunk 4 above)
Comment (wffurr) [reply types/resize-observer-browser/index.d.ts]: If I remove "declare" from the class declaration, I get the following error:

index.d.ts:11:1 - error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.

Comment (wffurr) [reply types/resize-observer-browser/index.d.ts]: All of the handbook examples include declare on namespace, class, and var.

File: types/resize-observer-browser/index.d.ts (hunk 1 above)
Comment (wffurr) [reply types/resize-observer-browser/index.d.ts]: Done.

File: types/resize-observer-browser/test/resize-observer-browser-module.tests.ts (hunk 3 above)
Comment (wffurr) [reply types/resize-observer-browser/test/resize-observer-browser-module.tests.ts]: Done.

File: types/resize-observer-browser/test/resize-observer-browser-module.tests.ts (hunk 2 above)
Comment (wffurr) [reply types/resize-observer-browser/test/resize-observer-browser-module.tests.ts]: Done.

File: types/resize-observer-browser/index.d.ts (hunk 4 above)
Comment (chivesrs) [reply types/resize-observer-browser/index.d.ts]: Huh, ok. I thought `.d.ts` implied declare automatically. Guess I was wrong.

File: types/resize-observer-browser/resize-observer-browser-tests.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+function resizeObserverCreates(): void {
+    const resizeObserver: ResizeObserver = new ResizeObserver(entries => {
```
Comment (chivesrs) [inline types/resize-observer-browser/resize-observer-browser-tests.ts]: nit: could you make the parameter `(entries)` like it was before?

**Review by chivesrs — APPROVED** [review]

File: types/resize-observer-browser/resize-observer-browser-tests.ts (hunk 5 above)
Comment (wffurr) [reply types/resize-observer-browser/resize-observer-browser-tests.ts]: Done.

I wish github would show this as a move.  `git show` shows it as a rename + changes.

Comment (wffurr) [reply types/resize-observer-browser/resize-observer-browser-tests.ts]: Now it shows the move + diff!  I don't know what I did.

**Pull request merged**